org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/PreMain.java

```diff
@@ -58,58 +55,23 @@ public static void premain(final String options, final Instrumentation inst)
 	private static IRuntime createRuntime(final Instrumentation inst)
```

Comment:
To further encapsulate the implementation details of `InjectedClassRuntime` I would add a static method `isSupported()` to this class. The code here in PreMain then reduces to:

	private static IRuntime createRuntime(final Instrumentation inst)
			throws Exception {
		if (InjectedClassRuntime.isSupported()) {
			return InjectedClassRuntime.createFor(inst);
		} else {
			return ModifiedSystemClassRuntime.createFor(inst,
					"java/lang/UnknownError");
		}
	}